### ISSUE TITLE
Contradiction in xml_set_default_handler() docs

### ISSUE THREAD

**Issue by php4fan** [php/doc-en#1391]
From manual page: https://php.net/function.xml-set-default-handler

It says:
> Description
> xml_set_default_handler([XMLParser](https://www.php.net/manual/en/class.xmlparser.php) $parser, [callable](https://www.php.net/manual/en/language.types.callable.php) $handler): bool

Ok so `$handler` is a callable. We all know what a callable is: either a string which is the name of a function,  an `[$object, 'methodName']` pair, or an anonymous function definition.

But then it says:
> handler
> **handler is a string** containing **the name** of a function that must exist when [xml_parse()](https://www.php.net/manual/en/function.xml-parse.php) is called for parser.

Wait, so must it be a string? Or is this just a leftover from an older version when callables didn't exist and it actually had to be a string with a function name?

**Comment by damianwadley** [comment]
Looks like a leftover: I don't see it requiring a string, just a generic array/object/stringable value.

But I do see that it accepts _any_ array/object/stringable value for that handler, meaning it hasn't switched to more modern support that only allows valid callbacks. https://3v4l.org/0kBnZ

**Comment by cmb69** [comment]
Yeah, [that code](https://github.com/php/php-src/blob/master/ext/xml/xml.c#L432-L439) could need some TLC.

**Comment by Girgias** [comment]
I'm going to look into the C code to improve it and will handle the docs at the same time.

**Comment by nielsdos** [comment]
Similarly, [xml_set_element_handler](https://www.php.net/manual/en/function.xml-set-element-handler) says stuff like "The function named by <parameter>start_handler</parameter>", which is also slightly inaccurate.
